Added Scale 2 (__guess_ticker_fallback) pdf parser functionallity

ItayShmuel opened the pull request:
scans company names for all pages of the pdf +
makes sure that the ticker is also appears in the pdf
if one of the conditions is not met, function returns none

roihala committed:
Telegram bot is stateful

roihala committed:
Devided bot class

roihala committed:
Better start message

roihala committed:
Before rebase

roihala committed:
Added survey

ItayShmuel committed:
Added Scale 2 (__guess_ticker_fallback) pdf parser functionallity

scans company names for all pages of the pdf +
makes sure that the ticker is also appears in the pdf
if one of the conditions is not met, function returns none

ItayShmuel committed:
Added Scale 2 (__guess_ticker_fallback) pdf parser functionallity

scans company names for all pages of the pdf +
makes sure that the ticker is also appears in the pdf
if one of the conditions is not met, function returns none

roihala committed:
Added location and finished refactor

roihala committed:
Merge pull request #339 from roihala/refactor/state_bot

Refactor/state bot

roihala committed:
Much better daily alerter

roihala committed:
Merge pull request #340 from roihala/fix/daily_alerter

Much better daily alerter

roihala committed:
Updating record_id again

roihala committed:
Merge pull request #341 from roihala/fix/records_collector

Updating record_id again

roihala committed:
Filings alerts back at 6 months

roihala committed:
disable scheduler logs

roihala committed:
Added pdfs dir

roihala commented on src/collect/records/dynamic_records_collector.py:
```diff
@@ -39,6 +39,11 @@ def collect(self):
+
+                # Scale 2 / Fallback (scan all pages + ticker appears in the pdf)
+                if not ticker:
+                    ticker = self.__guess_ticker_fallback(record_id, response)
```
We shouldn't treat this as a fallback,
put this logic in __guess_ticker #TODO

roihala commented on src/collect/records/dynamic_records_collector.py:
```diff
@@ -93,16 +98,28 @@ def __get_responses(self):
             if ticker:
                 return ticker
             else:
                 # TODO: Other guess
```
Call scale_2 function here, and give it more informative name.

This will be the place to call all scale functions

roihala commented on src/collect/records/dynamic_records_collector.py:
```diff
@@ -93,16 +98,28 @@ def __get_responses(self):
     def __guess_ticker(self, record_id, response):
         with fitz.open(self.get_pdf(record_id, response)) as doc:
-            ticker = self.__guess_by_company_name(doc)
+            ticker = self.__guess_by_company_name(doc, 3)
```
Either document what this 3 is for or export to CONST

roihala commented on src/collect/records/dynamic_records_collector.py:
```diff
@@ -93,16 +98,28 @@ def __get_responses(self):
 
-    def __guess_by_company_name(self, doc):
-        for page_number in range(0, MAX_PAGE_SEARCH):
+    def __guess_ticker_fallback(self, record_id, response):
```
This function still returns a single value,
In scale 1 - use the result it it contains single value,
In scale 2 - find if any of those values appear as ticker

roihala commented on src/collect/records/dynamic_records_collector.py:
```diff
@@ -93,16 +98,28 @@ def __get_responses(self):
-    def __guess_by_company_name(self, doc):
-        for page_number in range(0, MAX_PAGE_SEARCH):
+    def __guess_ticker_fallback(self, record_id, response):
+        with fitz.open(self.get_pdf(record_id, response)) as doc:
```
This shouldn't happen twice.
Relates to line 45 comment

roihala commented on src/collect/records/dynamic_records_collector.py:
```diff
@@ -146,6 +164,18 @@ def __extract_company_names_from_pdf(self, doc, page_number: int) -> list:
+
     def __get_symbol_from_map(self, comp_name: str) -> str:
         # TODO: Change names.csv company names to lower without commas or dots.
         # that will save us the name.lower().replace(',', '').replace('.', '') statement.
```
The replace still exists

roihala committed:
Merge pull request #342 from roihala/fix/records_collector

Fix/records collector

roihala committed:
Supporting update of trial source

roihala committed:
exception handling on fitz import

roihala committed:
Renamed img to images

roihala committed:
Fixed record alerter

roihala committed:
Merge pull request #349 from roihala/fix/records_alerter

Fixed record alerter

roihala committed:
Shuffeled alerted users

roihala committed:
Small tweaks

roihala committed:
Added stocker date

roihala committed:
Merge pull request #359 from roihala/feat/stocker_date

Added stocker date

roihala committed:
Broadcast quick fix

roihala committed:
Added  deregister link

roihala committed:
Changed agreemant url

roihala committed:
Merge pull request #367 from roihala/feat/deregister

Feat/deregister

roihala committed:
Removed _write flag

roihala committed:
Separated Runnable's cli args

roihala committed:
Not droping drop_keys from db anymore

nadavkluger committed:
Removed location from survey bot

roihala committed:
Added TA alerts

roihala committed:
Better exception handling

roihala committed:
Not erasing important users data anymore

roihala committed:
Created a link for mambo

roihala committed:
Added all_tickers csv

ItayShmuel committed:
Changed the way ticker gets picked with score calculation

+ guess ticker by mail, url and phone number
+ regex constants that gets compiled (saves time)

ItayShmuel committed:
Changed the way ticker gets picked with score calculation

+ guess ticker by mail, url and phone number
+ regex constants that gets compiled (saves time)

roihala committed:
Quick fix

roihala committed:
Merge pull request #370 from roihala/fix/remove_location

Removed location from survey bot

ItayShmuel committed:
fix Certificates error

fix No module named 'frontend' error (requirements.txt)

ItayShmuel committed:
fix Certificates error

fix No module named 'frontend' error (requirements.txt)

ItayShmuel committed:
Fix mongoDB query bugs (wrong collection)

Added find symbol by zip codes
Changed a bit Score calculations

ItayShmuel committed:
Added Symbols blacklists

ItayShmuel committed:
reslove conflicts
